spec/cb/utils/validator_spec.rb

```diff
@@ -12,127 +12,125 @@
 
-    it 'should return empty hash when body is improper json/xml' do
-      allow(response.response).to receive(:body).and_return('json')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    it 'should return empty hash when response is nil' do
-      response = nil
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
-    end
+    context 'with an invalid body' do
+      let(:response_body) { 'json' }
 
-    it 'should raise DocumentNotFound error if url is invalid' do
-      allow(response.response).to receive(:body).and_return('<!DOCTYPE html></html>')
-      allow(response).to receive(:code).and_return 404
-      expect { ResponseValidator.validate(response) }.to raise_error(Cb::DocumentNotFoundError)
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    it 'should return a full json hash when response status is not 200 and content is json' do
-      allow(response.response).to receive(:body).and_return('{"TestJson":{"Test":"True"}}')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_falsey
-    end
+    context 'with a nil body' do
+      let(:response_body) { nil }
 
-    it 'should return a blank json hash when status is not 200 and content is html' do
-      allow(response.response).to receive(:body).and_return('<!DOCTYPE html></html>')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    context 'raises a ServiceUnavailableError' do
-      it 'when status code is 503' do
-        allow(response.response).to receive(:body).and_return('{"errors":[]}')
-        allow(response).to receive(:code).and_return 503
-        expect { ResponseValidator.validate(response) }.to raise_error(Cb::ServiceUnavailableError)
-      end
+    context 'with an unsuccessful response status' do
+      let(:response_code) { 500 }
+      let(:response_body) { '{"errors":["Something went terribly wrong."]}' }
 
-      it 'when status code is 500' do
-        allow(response.response).to receive(:body).and_return('{"errors":["Something went terribly wrong."]}')
-        allow(response).to receive(:code).and_return 500
-        expect { ResponseValidator.validate(response) }.to raise_error do |error|
+      it 'populates error message from the json response' do
+        expect { validation }.to raise_error do |error|
           expect(error.message).to eq '["Something went terribly wrong."]'
           expect(error).to be_a Cb::ServerError
         end
       end
 
-      it 'when status code is 500' do
-        allow(response.response).to receive(:body).and_return('{"Errors":["Something went terribly wrong."]}')
```

Comment:
You deleted this test where the context should be "When given an Errors node (with a capital E)" .. Sorry about the poor test name.